modules/lang-mustache/src/test/java/org/elasticsearch/script/mustache/SearchTemplateIT.java

```diff
@@ -106,6 +107,67 @@ public void testTemplateQueryAsEscapedString() throws Exception {
         assertThat(searchResponse.getResponse().getHits().getHits().length, equalTo(1));
     }
 
+    /**
+     * Test the template query's hit count with an index which has more than 10k documents, relates to #52801.
+     */
+    public void testTemplateQueryHitCount() throws Exception {
```

Comment:
Can you replace these tests with [yaml tests](https://github.com/elastic/elasticsearch/blob/master/modules/lang-mustache/src/test/resources/rest-api-spec/test/lang_mustache/30_search_template.yml). It's odd that the rest option is not used to test the behavior and we to prefer yaml tests when doing small integration tests like this.

Reply:
Maybe also add one for the _msearch_template since your pr fixes both.

Reply:
Can we define a index with more than 10K documents easily in a yaml test file? I'm not sure about that. This method just tests index with more than 10K documents,because the default value of `track_total_hits` is 10000, so we can always get an accurate hits count with index having less than 10000 documents.

Reply:
I don't think you need to index 10k documents to test this change. The test I'd expect is that setting `trackTotalHits` to a value other than `true/false` in a template search request should fail the entire request when used in combination with `rest_total_hits_as_int`.
This should be easy to add in a yml test. 
Regarding the test that should check that we track total hits by default if  `rest_total_hits_as_int` is set on search template request, I think a unit test of `TransportSearchTemplateAction#convert` would be enough. We shouldn't require to 10k docs for this.